### DISCUSSION TITLE
子组件使用引了context，导致Tabs切换会影响表单布局

### DISCUSSION THREAD

**Discussion by tpc-ht** [alibaba/formily#2395] [Q&A 2.x]
复现地址 ：[https://codesandbox.io/s/optimistic-wright-v4eqj?file=/App.tsx](https://codesandbox.io/s/optimistic-wright-v4eqj?file=/App.tsx)

**Comment by PlutoCA** [comment]
你这b组件用来FormLayout包了一层
![image](https://user-images.githubusercontent.com/31099607/140254865-1f9c1305-b200-4153-811f-a16fd27f5f17.png)

**Reply by tpc-ht** [reply]
即使不用FormLayout包裹。也还是会错位，a组件可以使用React.memo()解决。但是b组件会涉及到使用context中的变量。导致布局错误。

**Comment by tpc-ht** [comment]
曲线方案：独占一行的字段不被FormGrid包裹。可以暂时解决当前布局问题。
https://codesandbox.io/s/sleepy-elgamal-s6xhq?file=/App.tsx
